src/app/_shared/components/scrollbar/draggable.directive.ts

```diff
@@ -0,0 +1,77 @@
+import { Directive, ElementRef, Inject, Input, Output } from '@angular/core';
+import { DOCUMENT } from '@angular/common';
+import { fromEvent } from 'rxjs';
+import { map, switchMap, takeUntil } from 'rxjs/operators';
+
+import { ScrollbarComponent } from './scrollbar.component';
+
+export type TypedMouseEvent<T extends EventTarget> = MouseEvent & { target: T };
+
+function getOffsetVertical(
+  { clientY }: MouseEvent,
+  { top, height }: ClientRect,
+): number {
+  return (clientY - top) / height;
+}
+
+function getOffsetHorizontal(
+  { clientX }: MouseEvent,
+  { left, width }: ClientRect,
+): number {
+  return (clientX - left) / width;
+}
+
+@Directive({
+  selector: '[appDraggable]',
+})
+export class DraggableDirective {
+  @Input()
+  draggable: 'vertical' | 'horizontal' = 'vertical';
+
+  @Output()
+  dragged = fromEvent<TypedMouseEvent<HTMLElement>>(
```

Comment:
Angular will never guarantee that EventEmitter will continue being an Observable. So don't use `@Output` this way. Declare this as EventEmitter and use `fromEvent().subscribe(dragged.emit)` in `ngOnInit()`